cmake/modules/hunter_download.cmake

```diff
@@ -382,6 +382,10 @@ function(hunter_download)
   file(REMOVE "${HUNTER_PACKAGE_HOME_DIR}/CMakeLists.txt")
   file(REMOVE "${HUNTER_DOWNLOAD_TOOLCHAIN}")
 
+  if(HUNTER_KEEP_PACKAGE_SOURCES OR HUNTER_${h_name}_KEEP_PACKAGE_SOURCES)
+    set(_hunter_keep_package_sources ON)
```

Comment:
Let's add
```cmake
else()
  set(_hunter_keep_package_sources OFF)
endif()
```
just to be sure.

Reply:
Changes applied [here](https://github.com/ruslo/hunter/pull/1216/commits/964c7fac0cada93077e51a22970d3f2fd4841afa)

The `HUNTER_${h_name}_KEEP_PACKAGE_SOURCES` is still getting cleared by the 3rd call to `hunter_config()` 

`grep "MOD: GTest" _logs/polly/libcxx/log.txt`

```
MOD: GTest
MOD: GTest  ON
MOD: GTest
```

which I believe comes from an evaluation of this call:

```
hunter_calculate_config_sha1.cmake:          "hunter_config(${x} "
```

Reply:
> The HUNTER_${h_name}_KEEP_PACKAGE_SOURCES is still getting cleared by the 3rd call

It's not cleared - it wasn't set in scope we need (there is no `set(HUNTER_..._SOURCES "")` command or `unset(HUNTER_..._SOURCES)`).

TL;DR Use global property: https://github.com/ruslo/hunter/commit/c82aed035c3db6813d29306cc9e5b1f5b1a1fa06

```
MOD: GTest
MOD: GTest  ON
MOD: GTest
```

We are processing config 3 times:
* global default config: https://github.com/ruslo/hunter/blob/353628aee9e834e72c82ae9377c7fff019a11327/cmake/modules/hunter_calculate_config_sha1.cmake#L28-L30
* local config: https://github.com/ruslo/hunter/blob/353628aee9e834e72c82ae9377c7fff019a11327/cmake/modules/hunter_calculate_config_sha1.cmake#L36-L38
* unified config (merge of two configs which is used to calculate config-id and forwarding for all 3rd parties): https://github.com/ruslo/hunter/blob/353628aee9e834e72c82ae9377c7fff019a11327/cmake/modules/hunter_finalize.cmake#L136-L138

First two processed inside function so user's scope is not affected. We need something that will outlive local scope. It can be chain of `PARENT_SCOPE`, cache variables or properties. Global property fits nicely here.

Reply:
Very helpful.  Thanks.